mods/hv/maps/scientist-rescue/scientist-rescue.lua

```diff
@@ -0,0 +1,116 @@
+--[[
+   Copyright 2024 The OpenHV Developers (see AUTHORS)
+   This file is part of OpenHV, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+
+LabGuardsTeam = { LabGuard1, LabGuard2, LabGuard3, LabGuard4, LabGuard5, LabGuard6 }
+ExtractionPath = { DropshipSpawnPoint.Location, RescuePoint.Location }
+InsertionPath = { EntryPoint.Location, ShorePoint.Location }
+PassengerReinforcements = { "mgpod", "mgpod", "electricpod", "electricpod", "mortarpod", "mortarpod", "flamepod" }
+OpeningAttack = { Ship1, Ship2, Ship3, Ship4 }
+
+
+Warning = UserInterface.Translate("warning")
+
+SendInsertionNavalTransport = function()
+	local passengers = Reinforcements.ReinforceWithTransport(Human, "navaltransport", PassengerReinforcements, InsertionPath)[2]
+	Trigger.OnAllKilled(passengers, RescueFailed)
```

Comment:
```suggestion
	Trigger.OnAllKilled(passengers, function()
		if not Human.IsObjectiveCompleted(KillGuardsObjective) then
			RescueFailed()
		end
	end)
```
If the ship and scientist have spawned, I wouldn't mind decoupling their death and mission failure. I will defer to your intent.